### PR TITLE
adds a helper method for creating symbols easier

### PR DESCRIPTION
**Pull request by ImanMh**
Not much of a description here. Sounds like the old version could be written in a simpler manner.

### CODE REVIEW

**Comment by ImanMh** [comment]
I just don't know why ci fails. I get green tests on my local.

File: packages/shared/ReactSymbols.js [hunk 1]
```diff
@@ -10,37 +10,20 @@
 // The Symbol used to tag the ReactElement-like types. If there is no native Symbol
 // nor polyfill, then a plain number is used for performance.
 const hasSymbol = typeof Symbol === 'function' && Symbol.for;
```
[the remark this answers is not in the document]
Comment (ImanMh) [reply packages/shared/ReactSymbols.js]: doesn't ```typeof Symbol.for | false``` return "error undefined has no key for" for browsers that does not support symbol at all?

**Comment by gaearon** [comment]
Thanks for the PR, but we try to avoid stylistic changes like this. It adds an extra level of abstraction to something that has a clear purpose. By itself this change might be fine, but we've previously done too many changes like this and eventually it led to an over-abstracted codebase no one could understand. So we're trying to avoid extra layers like this now.

"Saving space" is a red herring. Make sure that you check the actual size difference in the compiled files—it's quite likely that either gzipping or GCC inlining can make the savings moot, or even make the "after" version bigger.

**Comment by ImanMh** [comment]
@gaearon Actually I didn't do it because of saving size, I thought it would be easier to use createSymbol function rather than testing hasSymbol over and over again. Even though it's not that much of an abstraction, isn't abstraction a tool we use to remove repetitive things?  The good thing about it is that the abstraction is local in this file, just like hasSymbol.

**Pull request closed without merging**